### PR TITLE
added emr-on-eks module and made it generic

### LINKED ISSUE #200: [FEATURE] Spark on EKS
**Is your feature request related to a problem? Please describe.**
ADAS customers are looking for a way kubernetes-native way to run Spark Applications, using Karpenter for autoscaling. We would like a demo-able Step Function using the EKS module to spark submit an application to K8s

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*
Fixes #200 

*Description of changes:*
- added emr-on-eks module and made it generic
- tested it on LocalZones
- added unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b31122656df68438b86d5a5858b671d01db13a67
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9085f0ca-4de7-45ef-b688-9ceedc937f5c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: a2c55aa7fc46f2d0e89f2b35d7c8e1021200bfc0
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=69ae0b9a-28a4-4148-b299-406689608d6e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b76ed821e456136e79631da8cf0c5a370e07ea48
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=183627fa-dccd-4bf0-afdb-e1fabf9d279b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: e0c1970b123631183871d1ab4466d1f93d3601da
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=2a10cef4-1ab3-4e73-9e60-9be979eafd26%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 3ab94ff5d80ede11e1c6bb1f3734b3f4bc762259
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=48e11449-54e5-4754-827f-d1a82c16a8d7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: config/dev-dgraeber.env [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export GLOBAL_CUSTOM_DEPLOYMENT_NAME_IDENTIFIER="dgraeber"
-export PRIMARY_ACCOUNT="1234567890"
\ No newline at end of file
+export PRIMARY_ACCOUNT="XXXXXXXXXX"
```
Comment (dgraeber) [inline config/dev-dgraeber.env:2]: we can probably remove this file entirely ;) ?

File: modules/compute/emr-on-eks/deployspec.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - |
+        eksctl get iamidentitymapping --cluster ${SEEDFARMER_PARAMETER_EKS_CLUSTER_NAME} --arn ${EMR_SERVICE_ACCOUNT_ROLE} \
+        && echo "IAM Identity Mapping already found" \
+        || eksctl create iamidentitymapping --cluster ${SEEDFARMER_PARAMETER_EKS_CLUSTER_NAME} --region ${AWS_REGION} --arn ${EMR_SERVICE_ACCOUNT_ROLE} --username emr-containers
+      # Un-Assume the EKS_CLUSTER_ADMIN_ROLE or the rest of the deploy will fail
+      - unset AWS_ACCESS_KEY_ID && unset AWS_SECRET_ACCESS_KEY && unset AWS_SESSION_TOKEN
+      - cdk deploy --require-approval never --progress events --app "python app.py" --outputs-file ./cdk-exports-emr.json ${SEEDFARMER_PROJECT_NAME}-${SEEDFARMER_DEPLOYMENT_NAME}-${SEEDFARMER_MODULE_NAME}
+      - python -c "import json; file=open('cdk-exports-emr.json'); print(json.load(file)['${SEEDFARMER_PROJECT_NAME}-${SEEDFARMER_DEPLOYMENT_NAME}-${SEEDFARMER_MODULE_NAME}']['metadata'])" > emr.json
+      # Merge 2 stacks outputs
+      - export SEEDFARMER_MODULE_METADATA=$(jq -sc '.[0] * .[1]' rbac.json emr.json)
```
Comment (dgraeber) [inline modules/compute/emr-on-eks/deployspec.yaml:26]: confirmed that the metadata output exists in SSM?

**Review by dgraeber — COMMENTED** [review]
LGTM - Minor comments

File: modules/compute/emr-on-eks/deployspec.yaml (hunk 2 above)
Comment (srinivasreddych) [reply modules/compute/emr-on-eks/deployspec.yaml:26]: yes :)

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: 9df4f0f72414d2fbd82209722fb0b8adae679eab
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b2138d04-b6d7-461f-86c8-0ebc20393bd3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b6ee84f44783669f2404a48fc225a79e529c5d0a
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=3e414ea5-aaa5-470e-a8b7-fe49ee26240e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: ba532656e6a984a3310a8969ec7309a6a090cd78
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=5315a7c8-1cac-4c55-9a79-88c649b5f353%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/compute/eks/stack.py [hunk 3]
```diff
@@ -571,13 +571,16 @@ def _create_cluster_admin_role(self, project_name, deployment_name, module_name,
...
             self,
             "ClusterAdminRole",
             role_name=f"{project_name}-{deployment_name}-{module_name}-{region}-masterrole",
-            assumed_by=iam.CompositePrincipal(
-                iam.AccountRootPrincipal(),
-                iam.ServicePrincipal("ec2.amazonaws.com"),
+            assumed_by=iam.PrincipalWithConditions(
+                iam.AccountPrincipal(account),
+                conditions={
+                    "ArnLike": {"aws:PrincipalArn": f"arn:aws:iam::{account}:role/{project_name}-{deployment_name}-*"}
```
Comment (kukushking) [inline modules/compute/eks/stack.py:582]: nit - allow other partitions

**Review by dgraeber — APPROVED** [review]

**Pull request merged**